nameko/rpc.py

```diff
@@ -363,11 +405,18 @@ def async(self, *args, **kwargs):
         )
         return self.call_async(*args, **kwargs)
 
+    def get_message_headers(self, worker_context, timeout=None):
+        headers = HeaderEncoder.get_message_headers(self, worker_context)
+        if timeout:
+            headers['x-message-ttl'] = timeout * 1000
```

Comment:
I've tried this key and the one that is mentioned in the RabbitMQ documentation and used by kombu (`expiration`) and it doesn't work.
Expected that when the expired message reaches the head of the queue it shouldn't be delivered but it is.

Reply:
Ok, I've got it working. Instead of setting it to the headers (app specific) we need to set the property on the message when we publish it.

```
class _TimeoutAwareMethodProxy(nameko.rpc.MethodProxy):
    """Timeout aware RPC method proxy."""

    def __init__(
            self, *, timeout_secs: float, **kwargs,
    ) -> None:
        super().__init__(**kwargs)
        self.publisher.expiration = timeout_secs
```